modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -493,9 +523,11 @@
                                                    :dest-column-name (.getString rs "PKCOLUMN_NAME")}))))
      (catch SnowflakeSQLException e
        ;; dynamic tables doesn't support kks so it's fine to suppress the exception
-       (if (= "DYNAMIC_TABLE"
-              (-> (.getTables metadata db-name-or-nil schema-name table-name nil)
-                  jdbc/result-set-seq first :table_type))
+       (if (some?
+            (first (jdbc/query {:connection conn}
+                               (show-command-sql
+                                "DYNAMIC TABLES"
+                                :database db-name-or-nil :schema schema-name :like table-name))))
```

Comment:
Based on the implemenation of `show-command-sql`, providing `:database` and `:schema` should be mutually exclusive with each other. If a `:schema` is provided, then `:database` should not be provided and vice versa. If both are provided it would compile to `SHOW DYNAMIC TABLES IN DATABASE <database> SCHEMA <schema>` which is invalid. I think that should be in the docstring or preconditions of `show-command-sql` too, to prevent misuse.

Reply:
I've updatd to make it works for both cases